frontend/src/main/web/lib/components/glossary/DataTable.jsx

```diff
@@ -0,0 +1,485 @@
+    );
+  },
+
+  _getTransColumn: function() {
+    return (
+      <Column
+        label="Translations"
+        key={this.ENTRY.TRANS.field}
+        width={150}
+        dataKey={0}
+        flexGrow={1}
+        cellRenderer={this._renderTransCell}
+        headerRenderer={this._renderTransHeader}/>
+    );
+  },
+
+  _getPosColumn: function() {
+    return (
+      <Column
+        label="Part of Speech"
+        key={this.ENTRY.POS.field}
+        width={150}
+        dataKey={0}
+        cellRenderer={this._renderPosCell}
+        headerRenderer={this._renderPosHeader}/>
+    );
+  },
+
+  _getDescColumn: function() {
+    return (
+      <Column
+        label="Description"
+        key={this.ENTRY.DESC.field}
+        width={150}
+        flexGrow={1}
+        dataKey={0}
+        cellRenderer={this._renderDescCell}
+        headerRenderer={this._renderDescHeader}/>
+    );
+  },
+
+  _getTransCountColumn: function() {
+    return (
+      <Column
+        label="Translations"
+        key={this.ENTRY.TRANS_COUNT.field}
+        width={120}
+        cellClassName="tac"
+        dataKey={0}
+        cellRenderer={this._renderTransCountCell}
+        headerRenderer={this._renderTransCountHeader}/>
+    );
+  },
+
+  _getActionColumn: function() {
+    return (
+      <Column
+        label=""
+        key="Actions"
+        cellClassName="ph1/4"
+        width={300}
+        dataKey={0}
+        isResizable={false}
+        cellRenderer={this._renderActionCell}/>
+    )
+  },
+
+  _onRowMouseEnter: function (event, rowIndex) {
+    if (this.state.hoveredRow !== rowIndex) {
+      this.setState({hoveredRow: rowIndex});
+    }
+  },
+
+  _onRowMouseLeave: function () {
+    if (this.state.hoveredRow !== this.NO_ROW) {
+      this.setState({hoveredRow: this.NO_ROW});
+    }
+  },
+
+  _onRowClick: function (event, rowIndex) {
+    var resId = this._rowGetter(rowIndex)[this.resIdIndex];
+    if(this.props.focusedRow.rowIndex !== rowIndex) {
+      Actions.updateFocusedRow(resId, rowIndex);
+    }
+  },
+
+  _rowClassNameGetter: function (rowIndex) {
+    if(this.props.focusedRow && this.props.focusedRow.rowIndex === rowIndex) {
+      return 'bgcsec30a cdtrigger';
+    } else if(this.state.hoveredRow === rowIndex) {
+      return 'bgcsec20a cdtrigger';
+    }
+  },
+
+  _getGlossaryEntry: function (resId) {
+    return this.props.glossaryData[resId];
+  },
+
+  /**
+   * returns resId in list for glossary entry.
+   * Used for fixed-data-table when loading each row. See {@link _getGlossaryEntry}
+   * @param rowIndex
+   * @returns [resId] - resId in list
+   */
+  _rowGetter: function(rowIndex) {
+    var self = this,
+      row = this.props.glossaryResId[rowIndex];
+    if(_.isUndefined(row) || row === null) {
+      if(this.state.timeout !== null) {
+        clearTimeout(self.state.timeout);
+      }
+      this.state.timeout = setTimeout(function() {
+        Actions.loadGlossary(rowIndex);
+      }, self.TIMEOUT);
```

Comment:
> > No need for self here, just use this

> Need for self.Timeout

That is not within the function scope so still not needed.

If it were in the function scope, you could use a `() =>` function instead. I use arrow functions for most callbacks these days so that I don't need to worry about that.